File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i, x in enumerate(nums):
            table[x] = i
        
        for i, x in enumerate(nums):
            if (target-x) in table and table[target-x]!=i:
                return [min(i, table[target-x]), max(i, table[target-x])]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l=1, r=numbers.length;
        int[] res= new int[2];
        while(l<r){
            if((numbers[l-1]+numbers[r-1])==target){
                res[0] = l;
                res[1] = r;
                return res;
            }else if((numbers[l-1]+numbers[r-1])<target){
                l+=1;
            }else{
                r-=1;
            }
        }
        return res;
    }
}
